CONTRIBUTING.md

```diff
@@ -0,0 +1,51 @@
+# Contributing to the CodeFlare Operator
+
+Here are a few things to go over before getting started with CodeFlare Operator development:
+
+## Environment setup
+
+The following should be installed in your working environment:
+installation/)
+ - Go 1.18.X
+   - [Download release](https://go.dev/dl/)
+   - [Install Instructions](https://go.dev/doc/install)
+ - [Operator SDK](https://sdk.operatorframework.io/docs/installation/)
+ - GCC
+
+## Basic Overview
+Under the `api` dir, the MCAD and InstaScale custom resources are defined:
+ - See `mcad_types.go` and `instascale_types.go`
+
+Under `config/internal` are where the MCAD and InstaScale resource templates can be found:
+ - Sorted under `mcad` and `insascale` subdirs
+
+The code for MCAD/InstaScale resource reconsilliation can be found in the `controllers` dir:
+ - See `mcad_controller.go` and `instascale_controller.go`
+
+The main entrypoint for the operator is `main.go`
+
+## Building and Deployment
+If changes are made in the `api` dir, run: `make manifests`
+ - This will generate new CRDs and associated files
+
+If changes are made to any Go code (like in the `controllers` dir for example), run: `make`
+ - This will check and build/compile the modified code
+
+For building and pushing a new version of the operator image:
```

Comment:
Is this something you want developers to do? Is the intention that they would be pushed to their own dev image repos?

Reply:
I think in this case it's something developers would need to do for their own repos